opb-crank/src/scripts/crank.ts

```diff
@@ -0,0 +1,227 @@
+
+      if (events.length === 0) {
+        continue;
+      }
+
+      const accounts: Set<string> = new Set();
+      for (const event of events) {
+        accounts.add(event.openOrders.toBase58());
+
+        // Limit unique accounts to first 10
+        if (accounts.size >= maxUniqueAccounts) {
+          break;
+        }
+      }
+
+      const openOrdersAccounts = [...accounts]
+        .map((s) => new PublicKey(s))
+        .sort((a, b) => a.toBuffer().swap64().compare(b.toBuffer().swap64()));
+
+      // coinFee & pcFee are redundant for cranking.
+      // Instead, we pass spotMarkets[i]['_decoded'].eventQueue
+      // using duplicate accounts will reduce transaction size
+      const instr = DexInstructions.consumeEvents({
+        market: spotMarkets[i].publicKey,
+        eventQueue: spotMarkets[i].decoded.eventQueue,
+        coinFee: spotMarkets[i].decoded.eventQueue,
+        pcFee: spotMarkets[i].decoded.eventQueue,
+        openOrdersAccounts,
+        limit: consumeEventsLimit,
+        programId: serumProgramId,
+      });
+
+      crankInstructionsQueue.push(instr);
+
+      // if the queue is large then add the priority fee
+      if (events.length > priorityQueueLimit) {
+        instructionBumpMap.set(instr, 1);
+      }
+
+      // bump transaction fee if market address is included in PRIORITY_MARKETS env
+      if (priorityMarkets.includes(spotMarkets[i].publicKey.toString())) {
+        instructionBumpMap.set(instr, 1);
+      }
+
+      log.info(`market ${spotMarkets[i].publicKey} creating consume events for ${events.length} events`);
+    }
+
+    // send the crank transaction if there are markets that need cranked
+    if (crankInstructionsQueue.length > 0) {
+      // chunk the instructions to ensure transactions are not too large
+      const chunkedCrankInstructions = chunk(crankInstructionsQueue, maxTxInstructions);
+
+      chunkedCrankInstructions.forEach((transactionInstructions) => {
+        const crankTransaction = new Transaction({ ...recentBlockhash });
+
+        crankTransaction.add(
+          ComputeBudgetProgram.setComputeUnitLimit({
+            units: (CuLimit * maxTxInstructions),
+          }),
+        );
+
+        crankTransaction.add(...transactionInstructions);
+
+        crankTransaction.sign(payer);
+
+        // send the transaction
+        connection.sendRawTransaction(crankTransaction.serialize(), {
+          skipPreflight: true,
+          maxRetries: 2,
+        }).then((txId) => log.info(`Cranked ${transactionInstructions.length} market(s): ${txId}`));
+      });
+    }
+  } catch (e) {
+    if (e instanceof Error) {
+      switch (e.message) {
+        case 'Minimum context slot has not been reached':
+          // lightweight warning message for known "safe" errors
+          log.warn(e.message);
+          break;
+        default:
+          log.error(e);
+      }
+    }
+  }
+  await sleep(interval * 1000);
+  trytoCrank(spotMarkets);
```

Comment:
Does this run into recursion depth issues? We can restructure, but if it doesnt, dont worry about it.

Reply:
Not sure, crank has been running as desired locally for ~5hrs